pandas/core/categorical.py

```diff
@@ -2161,12 +2165,19 @@ class CategoricalAccessor(PandasDelegate, PandasObject, NoNewAttributesMixin):
 
     """
 
-    def __init__(self, values, index, name):
-        self.categorical = values
-        self.index = index
-        self.name = name
+    def __init__(self, data):
+        self._validate(data)
```

Comment:
see my commet above, these can be non-private I think as these are not user exposed